src/derivative_operators/derivative_operator.jl

```diff
@@ -85,12 +103,13 @@ function UpwindDifference{N}(derivative_order::Int,
                           approximation_order::Int, dx::T,
                           len::Int, coeff_func=nothing) where {T<:Real,N}
 
-    stencil_length          = derivative_order + approximation_order - 1 + (derivative_order+approximation_order)%2
+    stencil_length          = derivative_order + approximation_order
     boundary_stencil_length = derivative_order + approximation_order
-    dummy_x                 = -div(stencil_length,2) : div(stencil_length,2)
+    dummy_x                 = -1 : stencil_length - 2
     boundary_x              = -boundary_stencil_length+1:0
     boundary_point_count    = div(stencil_length,2) - 1 # -1 due to the ghost point
     # Because it's a N x (N+2) operator, the last stencil on the sides are the [b,0,x,x,x,x] stencils, not the [0,x,x,x,x,x] stencils, since we're never solving for the derivative at the boundary point.
+    @show dummy_x
```

Comment:
this

Reply:
When the stencil is not centred, it's length is different from the centred stencil. And since we want only 1 point on one side, I am starting `dummy_x` from `-1`.